GEOPM service sst integration test

cmcantalupo opened the pull request.

cmcantalupo committed:
Add a service test that uses the SST feature

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add automatically generated requirements.txt for geopmdpy

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Use rpm instead of yum or zypper when there is no network available

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fix formatting for raw signals in SSTIOGroup

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo commented on service/requirements.txt:
```diff
@@ -0,0 +1,4 @@
+cffi==1.13.2
```
This file needs to be accounted for in the MANIFEST_EXEMPT and the two Makefile.am EXTRA_DIST.

cmcantalupo committed:
Clean up inline python in sst test

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Clean up Makefiles and manifest to include new files

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add better error printing to sst test

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd commented on service/test/test_sst_priority.sh:
```diff
@@ -0,0 +1,106 @@
+CPU_FREQUENCY_STATUS core 6
+CPU_FREQUENCY_STATUS core 7' > ${READ_REQUEST}
+
+echo 'SST::COREPRIORITY_ENABLE:ENABLE board 0 1
```
Sorry, I omitted some steps in the runs I did earlier, since I had already configured them earlier in that session. You'll also want to set each of the weights and the frequency limits.

For buckets 0 through 3, something like:
```
SST::COREPRIORITY:<bucket>:WEIGHT board 0 <weight>
SST::COREPRIORITY:<bucket>:FREQUENCY_MIN board 0 <the starting frequency>
SST::COREPRIORITY:<bucket>:FREQUENCY_MAX board 0 <e.g. the system's max frequency>
```
Note that the buckets assigned here are defining the behavior that you map to cores with `SST::COREPRIORITY:ASSOCIATION`. That is, the `<bucket>` here is selected for a core by using that number as the input to `SST::COREPRIORITY:ASSOCIATION`.

For one configuration that would work for this test, you could set the min frequency to sticker for all except bucket 0. Maybe set bucket 0 to start somewhere inside the turbo range. Set a increasing weight as the bucket increases. You might consider bucket 0->0 weight, bucket 1->5 weight, bucket 2->10 weight, bucket 3->15 weight.

cmcantalupo commented on service/test/test_sst_priority.sh:
```diff
@@ -0,0 +1,105 @@
+    cores on the system are printed before and after configuring SST
+    core priority.
+
+    This test is a work in progress.
```
Remove this comment.

cmcantalupo committed:
Add more configuration options to SST test and add a sleep

- Addresses review comments and the sleep enables stress-ng to
  spin up all the CPUs.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>
